docs/chapters/incremental_indexing/index.md

```diff
@@ -0,0 +1,29 @@
+# Incremental Indexing in Jina
+
+## Generating Documents Ids:
+
+In Jina, the document ID is by default generated deterministically by computing a hash the document content. This avoids repeating the doc id while reindexing by using a Random or Incremental counter. The user can `override` document ids with Jina client and add `tags` to map to their unique concepts.
+
+## Incremental Indexing Design Pattern:
+
+The rationale behind Incremental indexing is to avoid re-indexing previously indexed documents, thus saving time and compute. This is done by persisting the document identifiers of indexed docs and using them as a DocIdCache.
+While indexing a document, it’s identifier is checked against the Cache for a ‘hit’ - if the document id exists,it’s already been indexed. We skip re-indexing it. If it’s a ‘miss’, we index it, adding the identifier to the Cache.
+1. DocIdCache stores doc ids in a int64 set and persists it to a numpy array
+2. UniqueVectorIndexer is a design pattern for combining VectorIndexer with DocIdCache
+3. UniquePbIndexer is a common design pattern for combining KVIndexer with DocIdCache
```

Comment:
The word design pattern makes me feel its a globally accepted design pattern. We can simply write "UniquePbIndexer combines KVIndexer with DocIdCache"